src/BDInfo/BDROM.cs

```diff
@@ -20,528 +20,421 @@
+        var pathFiles = directoryInfo.GetFiles();
+        var size = pathFiles.Where(pathFile => pathFile.Extension.ToUpper() != ".SSIF").Sum(pathFile => pathFile.Length);
+
+        var pathChildren = directoryInfo.GetDirectories();
+        size += pathChildren.Sum(GetDirectorySize);
+        return size;
+    }
+
+    public static int CompareStreamFiles(TSStreamFile x, TSStreamFile y)
+    {
+        // TODO: Use interleaved file sizes
+
+        if (x.FileInfo == null && y.FileInfo == null)
+        {
+            return 0;
+        }
 
-            return label;
+        if (x.FileInfo == null && y.FileInfo != null)
+        {
+            return 1;
         }
 
-        public static int CompareStreamFiles(
-            TSStreamFile x,
-            TSStreamFile y)
+        if (y.FileInfo == null && x.FileInfo != null)
         {
-            // TODO: Use interleaved file sizes
+            return -1;
+        }
 
-            if ((x == null || x.FileInfo == null) && (y == null || y.FileInfo == null))
-            {
-                return 0;
-            }
-            else if ((x == null || x.FileInfo == null) && y != null && y.FileInfo != null)
-            {
-                return 1;
-            }
-            else if ((x != null || x.FileInfo != null) && (y == null || y.FileInfo == null))
-            {
-                return -1;
-            }
-            else
-            {
-                if (x.FileInfo.Length > y.FileInfo.Length)
-                {
-                    return 1;
-                }
-                else if (y.FileInfo.Length > x.FileInfo.Length)
-                {
-                    return -1;
-                }
-                else
-                {
-                    return 0;
-                }
-            }
+        if (x.FileInfo.Length > y.FileInfo.Length)
+        {
+            return 1;
         }
 
-        private void ReadIndexVersion(Stream indexStream)
+        if (y.FileInfo.Length > x.FileInfo.Length)
         {
-            Span<byte> buffer = stackalloc byte[8];
-            int count = indexStream.Read(buffer);
-            int pos = 0;
-            if (count > 0)
-            {
-                var indexVer = ToolBox.ReadString(buffer, ref pos);
-                IsUHD = indexVer == "INDX0300";
-            }
+            return -1;
         }
 
-        [DllImport("kernel32.dll", CharSet = CharSet.Unicode, SetLastError = true)]
-        private static extern long GetVolumeInformation(
-            string PathName,
-            StringBuilder VolumeNameBuffer,
-            uint VolumeNameSize,
-            ref uint VolumeSerialNumber,
-            ref uint MaximumComponentLength,
-            ref uint FileSystemFlags,
-            StringBuilder FileSystemNameBuffer,
-            uint FileSystemNameSize);
+        return 0;
+    }
+
+    private void ReadIndexVersion(Stream indexStream)
+    {
+        var buffer = new byte[8];
+        var count = indexStream.Read(buffer, 0, 8);
```

Comment:
Go back to stackalloc for this small allocation

Reply:
I did so but also needed to re-add the helper function in toolbox